kornia/augmentation/_2d/intensity/plasma.py

```diff
@@ -0,0 +1,150 @@
+from typing import Dict, Optional, Tuple
+
+from torch import Size, Tensor
+from torch.distributions import Uniform
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.contrib import diamond_square
+
+
+class RandomPlasmaBrightness(IntensityAugmentationBase2D):
+    r"""Adds brightness to the image based on a fractal map generated by the diamond square algorithm.
+
+    .. image:: _static/img/RandomPlasmaBrightness.png
+
+    This is based on the original paper: TorMentor: Deterministic dynamic-path, data augmentations with fractals.
+    See: :cite:`tormentor` for more details.
+
+    .. note::
+        This function internally uses :func:`kornia.contrib.diamond_square`.
+
+    Args:
+        roughness: value to scale during the recursion in the generation of the fractal map.
+        intensity: value that scales the intensity values of the generated maps.
+        p: probability of applying the transformation.
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> img = torch.ones(1, 1, 3, 4)
+        >>> RandomPlasmaBrightness(roughness=(0.1, 0.7), p=1.)(img)
+        tensor([[[[0.6415, 1.0000, 0.3142, 0.6836],
+                  [1.0000, 0.5593, 0.5556, 0.4566],
+                  [0.5809, 1.0000, 0.7005, 1.0000]]]])
+    """
+
+    def __init__(
+        self, roughness: Tuple[float, float] = (0.1, 0.7), intensity: Tuple[float, float] = (0.0, 1.0), p: float = 0.5
+    ) -> None:
+        super().__init__(p=p)
+        self.roughness = Uniform(roughness[0], roughness[1])
+        self.intensity = Uniform(intensity[0], intensity[1])
+
+    def generate_parameters(self, shape: Size) -> Dict[str, Tensor]:
+        B = shape[0]  # batch_size
+        roughness = self.roughness.rsample((B,))
+        intensity = self.intensity.rsample((B, 1, 1, 1))
+        return dict(roughness=roughness, intensity=intensity)
```

Comment:
Simple uniform generations shall use `PlainUniformGenerator`

https://github.com/kornia/kornia/blob/3275944f20982be9cc777e34c455809cf59ba369/kornia/augmentation/_2d/intensity/sharpness.py#L56-L57

You may adjust the shape in the `apply_transform`

Reply:
I think we should refine this api part, because from user perspective is easier later to access to self.roughness and update with another distribution type. Also from developer perspective it’s faster instead of sometimes having to reimplement a generation. Or maybe, we need a better docs about how to use the low level api to create a new augmentations, what is needed etc

Reply:
It makes sense for this scenario since it is a too simple generation, but for some complicated ones like https://github.com/kornia/kornia/blob/3275944f20982be9cc777e34c455809cf59ba369/kornia/augmentation/random_generator/_2d/perspective.py. It gonna be a good practice to put all random parameter generation things together.

You do not need to reimplement a generation here, but to integrate the random parameter pre- and post- random generations into one. Users can now access `aug._param_genrator` to update the distributions. Later on, a strong `BasicGenerator` that accepts all different distributions is expected.

Reply:
done